src/components/App.js

```diff
@@ -1,21 +1,75 @@
 /* eslint-disable no-unused-vars */
 /* eslint-disable import/extensions */
+/* eslint-disable no-restricted-globals */
 
-import React from 'react';
+import React, { Component } from 'react';
 import './App.css';
 
 import ButtonPanel from './ButtonPanel';
 import Display from './Display';
 
 import calculate from '../logic/calculate.js';
 
-function App() {
-  return (
-    <div className="App">
-      <Display />
-      <ButtonPanel />
-    </div>
-  );
+class App extends Component {
+  constructor() {
+    super();
+    this.state = {
+      total: '',
+      next: '',
+      operation: null,
+      result: '',
+    };
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick(evt) {
+    const buttonName = evt.target.textContent;
+    const {
+      total, next, operation, result,
+    } = this.state;
+    if (operation === null && !isNaN(buttonName) && next === '') {
+      this.setState({
+        total: `${total + buttonName}`,
+        result: `${result + buttonName}`,
+      });
+    } else if (total !== '' && operation === null && buttonName !== '.') {
+      this.setState({
+        operation: buttonName,
+        result: `${`${result} ${buttonName} `}`,
+      });
+    } else if (!isNaN(buttonName)) {
+      this.setState({
+        next: `${next + buttonName}`,
+        result: `${result + buttonName}`,
+      });
+    }
+    if (buttonName === '=' || buttonName === 'AC' || buttonName === '+/-' || buttonName === '.') {
+      if (buttonName === 'AC') {
+        this.setState({ operation: buttonName });
+      } else if (buttonName === '+/-') {
+        this.setState({ operation: buttonName });
+      }
+      this.setState((prevState) => {
+        const result = calculate(prevState, buttonName);
+        return {
+          total: result.total,
+          next: result.next,
+          operation: result.operation,
+          result: buttonName === '=' ? result.total : result.result,
+        };
+      });
+    }
+  }
+
+  render() {
+    const { result } = this.state;
+    return (
+      <div className="App">
+        <Display result={result === '' ? '0' : result.toString()} />
```

Comment:
Let's fix this bug.

The `AC` button should clear the state. With the below screenshot when I click `=` button the app crashes. Same with the `-/+` button that needs to alternate the sign of the `next` and `total` value.

As I suggested earlier moving all the logic would help solve this more painlessly. You can check which button is clicked in the `calculate` function to know the object to return as the new state.

The `state` of the image below is:  `{total: -8, next: "0989", operation: "+/-", result: "1 AC 8 +/- 989"}`

![image](https://user-images.githubusercontent.com/36057474/71574464-15389e00-2ae9-11ea-8f10-bd40380eae8f.png)

Reply:
Thank you. Its now fixed.